### PR TITLE
Added new automatic job types.

### PR DESCRIPTION
**Pull request by stbof**
DOC-1428

### CODE REVIEW

**Review by irfansharif — APPROVED** [review]
The `AUTO SPAN CONFIG RECONCILIATION` bits look right to me.

**Review by maryliag — APPROVED** [review]
:lgtm:

> Reviewed 1 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/14785#-:-N99zpDxEdeSS014iV9F:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @stbof)

File: v22.1/show-jobs.md [hunk 1]
```diff
@@ -59,11 +59,11 @@ The following fields are returned for each job:
 Field | Description
 ------|------------
 `job_id` | A unique ID to identify each job. This value is used if you want to control jobs (i.e., [pause](pause-job.html), [resume](resume-job.html), or [cancel](cancel-job.html) it).
-`job_type` | The type of job. Possible values: `SCHEMA CHANGE`, [`BACKUP`](backup.html), [`RESTORE`](restore.html), [`IMPORT`](import.html), and [`CREATE STATS`](create-statistics.html). <br><br> For `SHOW AUTOMATIC JOBS`, the possible value is [`AUTO CREATE STATS`](cost-based-optimizer.html#table-statistics).
+`job_type` | The type of job. Possible values: `SCHEMA CHANGE`, [`BACKUP`](backup.html), [`RESTORE`](restore.html), [`IMPORT`](import.html), and [`CREATE STATS`](create-statistics.html). <br><br> For job types for automatic jobs, see [Show automatic jobs](#show-automatic-jobs).
```
Comment (taroface) [inline v22.1/show-jobs.md:62]:
```suggestion
`job_type` | The type of job. Possible values: `SCHEMA CHANGE`, [`BACKUP`](backup.html), [`RESTORE`](restore.html), [`IMPORT`](import.html), and [`CREATE STATS`](create-statistics.html). <br><br> For job types of automatic jobs, see [Show automatic jobs](#show-automatic-jobs).
```

File: v22.1/show-jobs.md [hunk 2]
```diff
@@ -119,12 +119,21 @@ You can filter jobs by using `SHOW JOBS` as the data source for a [`SELECT`](sel
...
+  786475982730133505 | AUTO SPAN CONFIG RECONCILIATION | reconciling span configurations                      |...
+  786483120403382274 | AUTO SQL STATS COMPACTION       | automatic SQL Stats compaction                       |...
+  786476180299579393 | AUTO CREATE STATS               | Table statistics refresh for movr.public.promo_codes |...
+...
+(8 rows)
 ~~~
 
+The job types of automatic jobs are:
+
+- `AUTO SPAN CONFIG RECONCILIATION`: a continuously running job that ensures that all declared [zone configurations](configure-zone.html) (`ALTER … CONFIGURE ZONE …`) are applied. For example, when `num_replicas = 7` is set on a table, the reconciliation job listens in on those changes and then informs the underlying [storage layer](architecture/storage-layer.html) to maintain 7 replicas for the table.
```
Comment (taroface) [inline v22.1/show-jobs.md:133]: I think we normally capitalize after colons that introduce list items, including elsewhere on this page:

```suggestion
- `AUTO SPAN CONFIG RECONCILIATION`: A continuously running job that ensures that all declared [zone configurations](configure-zone.html) (`ALTER … CONFIGURE ZONE …`) are applied. For example, when `num_replicas = 7` is set on a table, the reconciliation job listens in on those changes and then informs the underlying [storage layer](architecture/storage-layer.html) to maintain 7 replicas for the table.
```

File: v22.1/show-jobs.md [hunk 3]
```diff
@@ -119,12 +119,21 @@ You can filter jobs by using `SHOW JOBS` as the data source for a [`SELECT`](sel
...
+- `AUTO SQL STATS COMPACTION`: an hourly job that truncates the internal `system.statement_statistics` and `system.transaction_statistics` table row count to the value in the `sql.stats.persisted_rows.max` [cluster setting](cluster-settings.html). Both tables contribute to the [`crdb_internal.statement_statistics`](crdb-internal.html#statement_statistics) and [`crdb_internal.transaction_statistics`](crdb-internal.html#transaction_statistics) tables respectively.
```
Comment (taroface) [inline v22.1/show-jobs.md:134]:
```suggestion
- `AUTO SQL STATS COMPACTION`: An hourly job that truncates the internal `system.statement_statistics` and `system.transaction_statistics` table row counts to the value of the `sql.stats.persisted_rows.max` [cluster setting](cluster-settings.html). Both tables contribute to the [`crdb_internal.statement_statistics`](crdb-internal.html#statement_statistics) and [`crdb_internal.transaction_statistics`](crdb-internal.html#transaction_statistics) tables, respectively.
```

File: v22.1/show-jobs.md [hunk 4]
```diff
@@ -119,12 +119,21 @@ You can filter jobs by using `SHOW JOBS` as the data source for a [`SELECT`](sel
...
+- `AUTO CREATE STATS`: creates and updates [table statistics](cost-based-optimizer.html#table-statistics).
```
Comment (taroface) [inline v22.1/show-jobs.md:135]:
```suggestion
- `AUTO CREATE STATS`: Creates and updates [table statistics](cost-based-optimizer.html#table-statistics).
```

File: v22.1/show-jobs.md [hunk 5]
```diff
@@ -135,15 +144,16 @@ You can filter jobs by using `SHOW AUTOMATIC JOBS` as the data source for a [`SE
...
+...
+(7 rows)
 ~~~
 
 ### Show changefeed jobs
 
- You can display specific fields relating to changefeed jobs by running `SHOW CHANGEFEED JOBS`. These fields include:
+You can display specific fields relating to changefeed jobs by running `SHOW CHANGEFEED JOBS`. These fields include:
 
 * [`high_water_timestamp`](monitor-and-debug-changefeeds.html#monitor-a-changefeed): Guarantees all changes before or at this time have been emitted.
```
Comment (taroface) [inline v22.1/show-jobs.md:158]: Not part of your changes, but we use `-` rather than `*` in lists and it wouldn't hurt to change these opportunistically.

**Review by taroface — APPROVED** [review]
LGTM, with a few small nits.

**Pull request merged**